### ISSUE TITLE
Show layer feature for costume editor

### ISSUE THREAD

**Issue by ilbailba** [ScratchAddons/ScratchAddons#4067]
### Is your request related to a problem? Please describe.

**No**, it's not related to a problem, just requesting a new addon for Scratch Addons

### Describe the solution you'd like

It will show the **layers** in the **svg** editor in the **costume editor**. will appear as panels on the **right side** of the editor. It can be toggled from the eye icon and layers can be custom edited.

### Describe alternatives you've considered

eye icon **=>** _to hide or show the selected layer_

layer order **=>** _to change layer order of layers from top to bottom_

double click on the layer **=>** _only that layer will remain visible. other layers will not be visible._

### Additional context

More settings and tools can be added. I just wrote what I remember.
like the layers in **photoshop**

![sa](https://user-images.githubusercontent.com/71350431/149665140-bffa94c9-de2d-4019-b592-83a3f16e9345.png)

**Comment by lisa-wolfgang** [comment]
How would layers be initially defined? We should try to use editor data that's already there instead of trying to inject a property.

**Comment by ilbailba** [comment]
> How would layers be initially defined? We should try to use editor data that's already there instead of trying to inject a property.

Yeah. this is scratch's data. but it doesn't offer this to us and in multi-layer puppets this plugin can be very useful. For example, the data of the Onion skinning plug-in is scratch, but it is a plug-in.

**Comment by CST1229** [comment]
> > How would layers be initially defined? We should try to use editor data that's already there instead of trying to inject a property.
> 
> Yeah. this is scratch's data. but it doesn't offer this to us and in multi-layer puppets this plugin can be very useful. For example, the data of the Onion skinning plug-in is scratch, but it is a plug-in.

Like, is every layer one shape?

**Comment by LercDsgn** [comment]
> How would layers be initially defined? We should try to use editor data that's already there instead of trying to inject a property.

Not sure if provided by the API but a concept of layers in the costume editor exists as elements can be moved layers through the "Forward" and "Backward" buttons.

![image](https://user-images.githubusercontent.com/75680333/151561691-b8161cb0-07ce-4c9c-a095-3996688bb8b2.png)

**Comment by CST1229** [comment]
> > How would layers be initially defined? We should try to use editor data that's already there instead of trying to inject a property.
> 
> Not sure if provided by the API but a concept of layers in the costume editor exists as elements can be moved layers through the "Forward" and "Backward" buttons.
> 
> ![image](https://user-images.githubusercontent.com/75680333/151561691-b8161cb0-07ce-4c9c-a095-3996688bb8b2.png)

`addon.tab.redux.state.scratchPaint.undo.stack[addon.tab.redux.state.scratchPaint.undo.pointer]`)

(or in your browser console, `__scratchAddonsRedux.state.scratchPaint.undo.stack[__scratchAddonsRedux.state.scratchPaint.undo.pointer]`)

**Comment by ilbailba** [comment]
> > How would layers be initially defined? We should try to use editor data that's already there instead of trying to inject a property.
> 
> Not sure if provided by the API but a concept of layers in the costume editor exists as elements can be moved layers through the "Forward" and "Backward" buttons.
> 
> ![image](https://user-images.githubusercontent.com/75680333/151561691-b8161cb0-07ce-4c9c-a095-3996688bb8b2.png)
> 

Yeah. This is my main pillar.

**Comment by ilbailba** [comment]
> > > How would layers be initially defined? We should try to use editor data that's already there instead of trying to inject a property.
> > 
> > Yeah. this is scratch's data. but it doesn't offer this to us and in multi-layer puppets this plugin can be very useful. For example, the data of the Onion skinning plug-in is scratch, but it is a plug-in.
> 
> Like, is every layer one shape/a group of shapes or something else?

I don't understand exactly what you mean but I guess yes it is.

**Comment by ilbailba** [comment]
by the way i think group concept can be interpreted as folder when inside layers.
